main.go

```diff
@@ -98,9 +98,12 @@ func main() {
 	}
 	defer dbClient.Close(ctx)
 
+	reconcile := &reconciler.FleetReconciler{
+		// data required only once and is concurrent safe
+	}
 	reqChan := forman.GoForman(logger, conf.Reconcile.Concurrency,
 		func(req forman.Request) forman.Result {
-			return reconciler.Reconcile(logger, dbClient, req)
+			return reconciler.Reconcile(logger, reconcile, dbClient, req)
```

Comment:
This is wrong, every reconciliation should have its reconciler instance to hold the specific reconcile context.

Reply:
- There are some resources which are better instantiated once and re-used for every reconcile
- At the moment, those are AWS & IBM Client.

Reply:
- On another thought, I think I'm wrong due to not taking concurrency into account and sharing same struct with multiple workers🤦
- To be clean I don't see a way (yet) on using fields which can be instantiated once

Reply:
We already have that, good examples are logger and dbClient. They are instantiated once and used in every reconciliation invocation. The critical thing is that they need to be either stateless, immutable, or concurrency-safe.  
The point for the reconciler instance is to be a local context for a specific reconcile

Reply:
- yes, the FleetReconciler doesn't tick the requirements to be instantiated at once and so need to supply those via Reconcile function.

Reply:
done.